Server/middlewares/checkRequest.js

```diff
@@ -23,14 +23,49 @@ const requestCheck = (req, res, next) => {
   client.connect();
   client.query(queryString, (error, result) => {
     client.end();
-    if (result.rows[0]) {
+    if (!result.rows[0]) { return res.status(404).json({ status: 'error', code: 404, message: 'Request does not exist' }); }
+    const { status } = result.rows[0];
+    if (status === 'pending') {
+      return next();
+    }
+    return res.status(400)
+      .json({
+        status: 'error',
+        code: 400,
+        message: 'Request can no longer be approved or disapproved',
+      });
+  });
+};
+
+/**
+   * Checks if the request exists on the system and is approved
+   *
+   * @param {object} req - The request object received
+   * @param {object} res - The response object sent
+   * @param {object} next - The next middleware
+   */
+const requestCheckApprove = (req, res, next) => {
+  const { requestId } = req.params;
+  const queryString = {
+    text: 'SELECT * FROM requests WHERE id = $1 LIMIT 1;',
+    values: [requestId],
+  };
+  const client = new Client({
+    connectionString,
+  });
+  client.connect();
+  client.query(queryString, (error, result) => {
+    client.end();
+    if (!result.rows[0]) { return res.status(404).json({ status: 'error', code: 404, message: 'Request does not exist' }); }
+    const { status } = result.rows[0];
+    if (status === 'approved') {
```

Comment:
**Nitpick**
Is there a benefit by setting the status to approved as opposed to using a number for the status codes e.g. 1 for approved, 2 for rejected. I feel using this way it's a quicker process and it is not as tightly coupled.

Reply:
The numbering process is quicker, but when I retrieve the information from the database I would have to do checks and conversions on client or server in other to make it understandable to a person using the API.